druid-shell/src/platform/gtk/dialog.rs

```diff
@@ -27,9 +27,11 @@ pub(crate) fn get_file_dialog_path(
     options: FileDialogOptions,
 ) -> Result<OsString, Error> {
     // TODO: support message localization
-    let (title, action) = match ty {
-        FileDialogType::Open => ("Open File", FileChooserAction::Open),
-        FileDialogType::Save => ("Save File", FileChooserAction::Save),
+
+    let (title, action) = match (ty, options.select_directories) {
+        (FileDialogType::Open, false) => ("Open File", FileChooserAction::Open),
```

Comment:
Not a blocker, but personally I would probably write this as before (just matching ty) and then in the match arm I would do,

```rust
FileDialogOpen if options.select_directories => ("Open File", FileChooserAction::SelectFolder),
FileDialogOpen => ("Open File", FileChooserAction::Open),
...
```

Something that isn't clear to me from this is whether this will let you select a directory _or_ a file? Does that exist?

Reply:
From testing, it looks like you can only select folders, not files when using this method on gtk. 

This matches windows behaviour, so it seems correct?

Reply:
After looking at it a bit more, I think it should probably be titled "Open Folder" if you've asked for folders.

I've adjusted the matches arms bit to support this - let me know if you still think it'd be better restructured 😄 .